src/vs/workbench/electron-sandbox/window.ts

```diff
@@ -280,6 +294,29 @@ export class NativeWindow extends Disposable {
 		this.layoutService.updateWindowMaximizedState(maximized);
 	}
 
+	private getWindowMinimumWidth(panelPosition: Position = this.layoutService.getPanelPosition()): number {
+		// if panel and editor are both visible
+		// and panel is on the side, then return the larger minwidth
+		const panelAndEditorBothVisible = (!this.layoutService.isPanelHidden() && !this.layoutService.isPanelMaximized());
```

Comment:
In my testing, I was able to break the minimum width by moving the panel to the side, then using the mouse to snap the panel hidden. This might be fixed by using isVisible instead but it is something to investigate.

Reply:
Repro'd. It's not fixed by using isVisible instead. The issue is that when the user snaps the panel hidden, this function is called while the panel is still visible.
It could mean the event is firing at the wrong time.